homeassistant/components/nexia/util.py

```diff
@@ -16,3 +23,64 @@ def percent_conv(val):
     if val is None:
         return None
     return round(val * 100.0, 1)
+
+
+class SingleShot:
```

Comment:
It would be cleaner to move this helper into the `nexia` library itself. This keeps the integration logic encapsulated within the library, reducing maintenance overhead in Home Assistant and allowing for faster, independent updates without being tied to Home Assistant’s release cycle or internal code structure.

Also, consider replacing `async_call_later` with `loop.call_later` for a more direct approach.